### PR TITLE
EOS-24455: updated controlnode and storagenode logic in provisioner kubernetes deployment

### PR DESCRIPTION
**Pull request by tanujashinde0405**
# Problem Statement
- Problem statement

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

File: test/deploy/k8s/cortx-storagenode-statefulset.yml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            fieldRef:
+              fieldPath: metadata.name
+      affinity:
+        podAntiAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+          - topologyKey: kubernetes.io/hostname
+            labelSelector:
+              matchLabels:
+                app: storagenode
+      setHostnameAsFQDN: true
```
Comment (nitin-seagate) [inline test/deploy/k8s/cortx-storagenode-statefulset.yml:41]: `setHostnameAsFQDN: true` will not work with our standard kubectl version v1.19
Lets set hostname in container directly by copying value from `hostname --fqdn`

File: test/deploy/k8s/test.yaml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+nginx:
```
Comment (nitin-seagate) [inline test/deploy/k8s/test.yaml:1]: Why do we need this file?

File: test/deploy/k8s/cortx-controlnode-statefulset.yml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+      - name: cortx-secret
+        secret:
+          secretName: cortx-secret
+      - name: node-info
+        downwardAPI:
+          items:
+          - path: "name"
+            fieldRef:
+              fieldPath: metadata.name
+      setHostnameAsFQDN: true
```
Comment (nitin-seagate) [inline test/deploy/k8s/cortx-controlnode-statefulset.yml:34]: `setHostnameAsFQDN: true` will not work with our standard kubectl version v1.19
Lets set `hostname` in container directly by copying value from `hostname --fqdn`

File: test/deploy/k8s/deploy.sh [hunk 4]
```diff
@@ -1,21 +1,25 @@
 #!/bin/bash -x
 set -eu
 SCRIPT_DIR=$(dirname $0)
-"$SCRIPT_DIR"/destroy.sh
+"$SCRIPT_DIR"/destroy.sh || true
 kubectl create namespace cortx
 kubectl create cm solution-config --from-file="$SCRIPT_DIR"/cluster.yaml \
     --from-file="$SCRIPT_DIR"/config.yaml --namespace cortx
 
+kubectl apply -f "$SCRIPT_DIR"/cortx-storagenode-service.yml --namespace cortx
```
Comment (nitin-seagate) [inline test/deploy/k8s/deploy.sh:9]: Please restructure sequence:
- Control Node
- Storage Node

File: test/deploy/k8s/deploy.sh [hunk 5]
```diff
@@ -1,21 +1,25 @@
...
+kubectl apply -f "$SCRIPT_DIR"/cortx-controlnode-service.yml --namespace cortx
 kubectl apply -f "$SCRIPT_DIR"/cortx-secret.yml --namespace cortx
 kubectl apply -f "$SCRIPT_DIR"/cortx-pv-config.yml --namespace cortx
 kubectl apply -f "$SCRIPT_DIR"/cortx-pvc-config.yml --namespace cortx
-kubectl apply -f "$SCRIPT_DIR"/cortx-statefulset.yml --namespace cortx
+kubectl apply -f "$SCRIPT_DIR"/cortx-storagenode-statefulset.yml --namespace cortx
```
Comment (nitin-seagate) [inline test/deploy/k8s/deploy.sh:14]: Please restructure sequence:
- Control Node
- Storage Node

**Review by nitin-seagate — CHANGES_REQUESTED** [review]

**Review by kailashyadav11 — APPROVED** [review]

**Pull request merged**